completion/_catkin

```diff
@@ -53,7 +53,7 @@ _catkin_get_enclosing_workspace() {
       _workspace_root=$_path
       return 0
     else
-      _path="$(readlink -f $_path/..)"
+      _path="$(dirname "$_path")"
```

Comment:
I would assume that `readlink` was used on purpose. Imagine a workspace at `~/deep/in/the/file/system/workspace` and the `src` folder is linked to `~/src` for easier access. With your solution, the workspace would not be detected from within `~/src`. However we could change line 46 to `_path="$(readlink -f $1)"` to resolve all symbolic links and use `dirname` in the loop as you did.

Reply:
You are right; I haven't thought that somebody is using it in this way. I am jumping directly to a package using **roscd** command. Unfortunately, this doesn't solve my problem. We are using folder `~/git` as a place where all git repositories are cloned. Then, we are creating symlinks into the workspace. Therefore in the current version, the workspace is not found (`readlink` leads to `~/git`). For example for `catkin clean` command, the package suggestion doesn't work and I have to write the name of a package manually.

Reply:
Check the newest commit that solves my problem and also works for your example. The same functionality as before should be preserved.

Reply:
Yes, the updated version looks nice.